### PR TITLE
Specify product combinations in cart when there is not enough stock 

### LINKED ISSUE #23051: Specify product combinations in cart when there is not enough stock
#### Is your feature request related to a problem?

Improve the information displayed in the checkout when a product is without stock

From https://www.prestashop.com/forums/topic/1041442-nobody-can-figure-it-out-we-need-a-specialist/

#### Describe the solution you'd like

Display the combination from the product without stock.

#### Additional context

Which one is the product without stock?
image

### PR DESCRIPTION
**Pull request by JBWModules**
Fixes https://github.com/PrestaShop/PrestaShop/issues/23051

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Improve the information displayed in the checkout when a product is without stock<br>From https://www.prestashop.com/forums/topic/1041442-nobody-can-figure-it-out-we-need-a-specialist
| Type?             | improvement
| Category?         | FO 
| BC breaks?        |  no
| Deprecations?     | no
| Fixed ticket?     | Fixes #23051
| How to test?      | Create a cart with product with attribute. Remove stock and check cart again. Error message is appearing.
| Possible impacts? | Cart dispaly

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/23059)

### CODE REVIEW

**Comment by Progi1984** [comment]
@Julievrz Do you think we need to modify the translation when there are attributes ?

File: controllers/front/CartController.php [hunk 1]
```diff
@@ -637,7 +637,7 @@ protected function areProductsAvailable()
         if ($product['active']) {
             return $this->trans(
                 'The item %product% in your cart is no longer available in this quantity. You cannot proceed with your order until the quantity is adjusted.',
-                ['%product%' => $product['name']],
+                ['%product%' => $product['name'] . (isset($product['attributes']) ? ' ' . $product['attributes'] : '')],
```
Comment (matks) [inline controllers/front/CartController.php:640]: Hi @JBWModules I'm wondering, with this works with multilang products? Should'nt we pick the right localized strings, something that look like `$product['attributes'][$locale]` ?

Comment (JBWModules) [reply controllers/front/CartController.php:640]: It's already translated in the active lang (same as $product['name'] was before).

**Review by matks — DISMISSED** [review]

File: controllers/front/CartController.php (hunk 1 above)
Comment (matks) [reply controllers/front/CartController.php:640]: Thanks for the answer, code seems good for me 😊

**Review by Progi1984 — CHANGES_REQUESTED** [review]
Wait for Julie before approving

**Comment by Julievrz** [comment]
> @Julievrz Do you think we need to modify the translation when there are attributes ?

Yes I think attributes should be translated too.

File: controllers/front/CartController.php [hunk 2]
```diff
@@ -637,7 +637,7 @@ protected function areProductsAvailable()
         if ($product['active']) {
             return $this->trans(
                 'The item %product% in your cart is no longer available in this quantity. You cannot proceed with your order until the quantity is adjusted.',
```
Comment (Julievrz) [inline controllers/front/CartController.php]:
```suggestion
                'The product %product% in your cart is no longer available in this quantity. Please adjust the quantity to proceed with the order.',
```
This wording has been changed in another PR (https://github.com/PrestaShop/PrestaShop/pull/23093/files#diff-0dac05a6abe4d202319dd093b8ab2b37R654)

**Review by matks — DISMISSED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by Julievrz — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @JBWModules,

In the FO > Shopping cart page > the Message is not updated when we increase/descrease the quantity > quantity total.
We need to refresh the page.
This issue is reproduced with a simple product and a product with combinations.

**Steps to reproduce the issue**
1. Create a product with quantity = 3
2. Go to FO > Add the Product to the cart with quantity = 3 and Proceed to checkout
3. In the Shopping cart > Increase the quantity to 5
4. See error: the message is not updated
image
5. Refresh the page => ok
image
6. Decrease the quantity to 4
7. See error: the message is not updated
image

https://drive.google.com/file/d/1Js50uFovZ2KHNLxwtBfWNmY_iQxJ1gbx/view?usp=sharing

Thank you!

**Comment by PierreRambaud** [comment]
Hi @JBWModules ,

Since we had no news from you for more than 30 days, I'll close this pull request. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Pull request closed without merging**